frontend/src/components/dashboard/settings/Settings.tsx

```diff
@@ -50,6 +50,11 @@ const SETTINGS_LINKS_WITH_EMAIL = [
 
 const Settings = () => {
   const modalContext = useContext(ModalContext)
+  const [isLoading, setIsLoading] = useState(true)
+  const [fetchedUserSettings, setFetchedUserSettings] = useState(false)
+  const [fetchedCustomFromAddresses, setFetchedCustomFromAddresses] = useState(
+    false
+  )
```

Comment:
We don't need 3 different states to manage the loading for this one, just `isLoading` is enough and less confusing. We can do that by changing the `useEffect` on line 63-66 to
```ts
useEffect(() => {
  setIsLoading(true)
  Promise.all([
    fetchUserSettings(),
    fetchCustomFromAddresses(),
  ]).finally(() => setIsLoading(false))
}, [])
```

Reply:
oh shit that makes sense, let me fix that